### PR TITLE
feat: make jobs awaitable with asyncio, v1 (Proof of Concept)

### LINKED ISSUE #18: BigQuery: make jobs awaitable
I know BigQuery jobs are asynchronous by default. However, I am struggling to make my datapipeline async end-to-end.

Looking at [this JS example](https://medium.com/skyshidigital/what-bigquery-is-and-isnt-d092fa173d53), I thought it would be the most Pythonic to make a BigQuery job awaitable. However, I can't get that to work in Python i.e. errors when `await client.query(query)`. Looking at the source code, I don't see which method [returns an awaitable object](https://www.python.org/dev/peps/pep-0492/).

 I have little experience in writing async Python code and found [this example that wraps jobs in a `async def coroutine`](https://stackoverflow.com/questions/53065340/python-how-to-big-query-asynchronous-tasks).
```
class BQApi(object):                                                                                                 
    def __init__(self):                                                                                              
        self.api = bigquery.Client.from_service_account_json(BQ_CONFIG["credentials"])                               

    async def exec_query(self, query, **kwargs) -> bigquery.table.RowIterator:                                       
        job = self.api.query(query, **kwargs)                                                                        
        task = asyncio.create_task(self.coroutine_job(job))                                                          
        return await task                                                                                            

    @staticmethod                                                                                                    
    async def coroutine_job(job):                                                                                    
        return job.result()   
```

The [`google.api_core.operation.Operation`](https://googleapis.github.io/google-cloud-python/latest/core/operation.html) shows how to use `add_done_callback` to asynchronously wait for long-running operations. I have tried that, but the following yields `AttributeError: 'QueryJob' object has no attribute '_condition'` :

```
from concurrent.futures import ThreadPoolExecutor, as_completed
query1 = 'SELECT 1'
query2 = 'SELECT 2'

def my_callback(future):
    result = future.result()

operations = [bq.query(query1), bq.query(query2)]
[operation.add_done_callback(my_callback) for operation in operations]
results2 = []
for future in as_completed(operations):
  results2.append(list(future.result()))
```

Given that jobs are already asynchronous, would it make sense to add a method that returns an awaitable?

Or am I missing something and is there an Pythonic way to use the BigQuery client with the async/await pattern?

### PR DESCRIPTION
**Pull request by kiraksi**
This PR:
- Is a proof of concept test for the eventual AsyncQueryManager
- Should not be merged and will be closed after testing
- Successfully showed that async functionality works cohesively with our base code

Thank you for opening a Pull Request! Before submitting your PR, there are a few things you can do to make sure it goes smoothly:
- [X] Make sure to open an issue as a [bug/issue](https://github.com/googleapis/python-bigquery/issues/new/choose) before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea
- [X] Ensure the tests and linter pass
- [X] Code coverage does not decrease (if any source code was changed)
- [X] Appropriate docs were updated (if necessary)

Fixes #18 🦕

### CODE REVIEW

File: tests/unit/test_client.py [hunk 1]
```diff
@@ -9546,6 +9551,37 @@ def test_schema_to_json_with_file_object(self):
         client.schema_to_json(schema_list, fake_file)
         assert file_content == json.loads(fake_file.getvalue())
 
+    if sys.version_info >= (3, 9):
+
+        @patch("bigquery_client.async_query_and_wait", new_callable=AsyncMock)
+        async def test_async_execution(self, mock_query):
```
Comment (tswast) [inline tests/unit/test_client.py:9557]: Are we sure this test is detected and runnable by pytest? From https://stackoverflow.com/a/70016047/101923 it seems we may need to install another plugin for pytest and add a decorator.

Comment (kiraksi) [reply tests/unit/test_client.py:9557]: I did run the test without the if statement, it understandably caused errors for 3.7/3.8 for not being supported however passed for other python versions. It seems like the proof of concept works, but I will revisit how to run this test according to this example in the next commit!

Comment (kiraksi) [reply tests/unit/test_client.py]: Because I am using asyncio.to_thread, should I test if it returns a coroutine before the await, instead of a future?

**Comment by kiraksi** [comment]
The tests failing for this is for pytype, which is unimportant for the purpose of this PoC PR. Closing this as a successful demonstration of the viability of asynchrony in our library.

**Pull request closed without merging**